src/Http/Download.php

```diff
@@ -5,33 +5,30 @@
 namespace OpenEuropa\CdtClient\Http;
 
 use OpenEuropa\CdtClient\Contract\RestInterface;
-use OpenEuropa\CdtClient\Contract\TokenAwareInterface;
 use OpenEuropa\CdtClient\Exception\InvalidStatusCodeException;
+use OpenEuropa\CdtClient\Model\Response\Token;
 use OpenEuropa\CdtClient\Traits\SerializerAwareTrait;
-use OpenEuropa\CdtClient\Traits\TokenAwareTrait;
 use OpenEuropa\CdtClient\Traits\ValidationAwareTrait;
 
 /**
 * A class that download files from the API.
  *
- * @see TokenAwareInterface
  * @see ValidationAwareTrait
  * @see SerializerAwareTrait
 */
-class Download implements TokenAwareInterface
+class Download
 {
-    use TokenAwareTrait;
     use ValidationAwareTrait;
     use SerializerAwareTrait;
 
-    public function __construct(protected RestInterface $rest)
+    public function __construct(protected RestInterface $rest, protected Token $token)
     {
     }
 
     public function downloadFile(string $uri): string
     {
         try {
-            $response = $this->rest->get($uri, $this->getAuthorizationHeaders());
+            $response = $this->rest->get($uri, $this->rest->getAuthorizationHeadersFromToken($this->token));
```

Comment:
So I had to put this method somewhere outside the `AuthorisedEndpointBase`, it is also used by the `Download` class.

Reply:
y, I think the trait made sense from this point of view

Reply:
Eventually, I've added a smaller trait that provides this method